### ISSUE TITLE
Get two access logs when to pull an image

### ISSUE THREAD

**Issue by wy65701436** [goharbor/harbor#10253]
When I try to pull an image from harbor, harbor records two access_log because we get two notification from the distribution registry, the `harbor-registry-client` should be filtered out. 

Dec 12 22:11:35 172.30.0.1 core[19698]: 2019-12-13T06:11:35Z [DEBUG] [/core/service/notifications/registry/handler.go:252]: receive an event: 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----ID: 61ce2975-d960-43de-99c2-dc220ba17632 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----target: project1/busybox:latest 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----digest: sha256:679b1c1058c1f2dc59a3ee70eed986a88811c0205c8ceea57cec5f22d2c3fbb1 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----action: pull 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----mediatype: application/vnd.docker.distribution.manifest.v2+json 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----user-agent: `harbor-registry-client`
Dec 12 22:11:35 172.30.0.1 core[19698]: 2019/12/13 06:11:35 #033[1;44m[D] [server.go:2774] |     172.30.0.4|#033[42m 200 #033[0m|   5.112051ms|   match|#033[46m POST    #033[0m /service/notifications   r:/service/notifications#033[0m
Dec 12 22:11:35 172.30.0.1 core[19698]: 2019-12-13T06:11:35Z [DEBUG] [/core/filter/security.go:232]: OIDC CLI modifier only handles request by docker CLI or helm CLI
Dec 12 22:11:35 172.30.0.1 core[19698]: 2019-12-13T06:11:35Z [DEBUG] [/core/filter/security.go:450]: can not get user information from session
Dec 12 22:11:35 172.30.0.1 core[19698]: 2019-12-13T06:11:35Z [DEBUG] [/core/filter/security.go:505]: user information is nil
Dec 12 22:11:35 172.30.0.1 core[19698]: 2019-12-13T06:11:35Z [DEBUG] [/core/filter/security.go:519]: using local database project manager
Dec 12 22:11:35 172.30.0.1 core[19698]: 2019-12-13T06:11:35Z [DEBUG] [/core/filter/security.go:521]: creating local database security context...
Dec 12 22:11:35 172.30.0.1 core[19698]: 2019-12-13T06:11:35Z [DEBUG] [/core/service/notifications/registry/handler.go:252]: receive an event: 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----ID: 830d74ff-823d-454b-af2b-b830429bcf66 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----target: project1/busybox:latest 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----digest: sha256:679b1c1058c1f2dc59a3ee70eed986a88811c0205c8ceea57cec5f22d2c3fbb1 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----action: pull 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----mediatype: application/vnd.docker.distribution.manifest.v2+json 
Dec 12 22:11:35 172.30.0.1 core[19698]: ----user-agent: docker/19.03.2 go/go1.12.8 git-commit/6a30dfc kernel/4.15.0-58-generic os/linux arch/amd64 UpstreamClient(Docker-Client/19.03.2 \(linux\))

**Comment by reasonerjt** [comment]
Re-opening the issue as this is real case when using containerd to pull image form Harbor.

The access log in nginx looks like this:
```
remote_addr - "HEAD /v2/wcpharbor-143/ubuntu_sar/manifests/v1.10 HTTP/1.1" 200 0 "-" "containerd/1.2.10+unknown" 0.022 0.024 .
remote_addr - "GET /v2/wcpharbor-143/ubuntu_sar/manifests/sha256:0decf2bda7677075d705ef857f58cfbb0da39c476691f5058bc179ee690f0622 HTTP/1.1" 200 2416 "-" "containerd/1.2.10+unknown" 0.029 0.032 .
```

Unlike docker which GET the manifest directly.  containerd first sends a HEAD to the manifest with tag then sends a GET to the manifest with digest.  The registry sends two events for HEAD and GET.

That's why there are 2 access logs in the logs tab and the pull count increased by 2.

**Comment by reasonerjt** [comment]
Looks like the event has a reference to the request, maybe we can filter out the event that is a HEAD request, but the side effect is we lost the tag information in access log.

**Comment by reasonerjt** [comment]
Moving this out of SP76 as per discussion there is concern about the side effect.

**Comment by reasonerjt** [comment]
This should be fixed in 2.0 as we rework the notification mechanism

**Comment by reasonerjt** [comment]
Verified there will be only one log in the master build, closing.
